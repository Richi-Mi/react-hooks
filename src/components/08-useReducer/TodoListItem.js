import React from 'react';
import PropTypes from 'prop-types';

const TodoListItem = ({ id, desc, index, done, handleDelete, handleToggle }) => {
    return (
        <li 
            key={id}
            className='list-group-item'
        >
            <p 
                onClick={() => handleToggle(id)} 
                className={`${done && 'complete'}`}
            > 
                {index + 1}. - {desc}  
            </p>

            <button 
                className='btn btn-danger' 
                onClick={() => handleDelete(id)}
            >
                Borrar 
            </button>
        </li>
    )
}
TodoListItem.propTypes = {
    id: PropTypes.number.isRequired,
    desc: PropTypes.string, 
    done: PropTypes.bool.isRequired,
    index: PropTypes.number.isRequired, 
    handleDelete: PropTypes.func.isRequired, 
    handleToggle: PropTypes.func.isRequired
}

export default TodoListItem
