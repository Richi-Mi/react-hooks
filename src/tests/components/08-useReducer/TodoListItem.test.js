import { shallow } from "enzyme";
import TodoListItem from '../../../components/08-useReducer/TodoListItem';

describe(' - Pruebas en el componente <TodoListItem />', () => {
    const funcionDelete = jest.fn();
    const funcionToggle = jest.fn();
    const text_show = "Aprender a Testear";
    const id = 12234567891;
    const index = 0;
    let wrapper;

    beforeEach(() => {

        wrapper = shallow(
            <TodoListItem
                id={id}
                desc={text_show}
                index={index}
                done={false}
                handleDelete={funcionDelete}
                handleToggle={funcionToggle}
            />
        )
    })

    test(' - Debe de Mostrarse correctamente ', () => {

        expect( wrapper ).toMatchSnapshot()

    })
    test(' - Debe de llamar la función borrar ', () => {
        const btn_delete = wrapper.find('.btn');
        btn_delete.simulate('click')

        expect( funcionDelete ).toHaveBeenCalled();
        expect( funcionDelete ).toHaveBeenCalledTimes( 1 );
        // expect( funcionDelete ).toHaveBeenCalledWith( expect.any( Number ) );
        expect( funcionDelete ).toHaveBeenCalledWith( id );
    })
    test(' - Debe de llamar la función toggle ', () => {
        const p_text = wrapper.find('p');
        p_text.simulate('click');

        expect( funcionToggle ).toHaveBeenCalled();
        expect( funcionToggle ).toHaveBeenCalledTimes( 1 );
        expect( funcionToggle ).toHaveBeenCalledWith( id )
    })
    test(' - Debe de Mostrar el texto correctamente', () => {
        const p_text = wrapper.find('p');
        
        expect( p_text.text().trim() ).toBe(`${ index + 1 }. - ${text_show}`) 
    })
    test(' - Debe de tener la clase complete si el TODO.done es true', () => {
        const wrapper_done = shallow(
            <TodoListItem
                id={id}
                desc={text_show}
                index={index}
                done={true}
                handleDelete={funcionDelete}
                handleToggle={funcionToggle}
            />
        )
        const p_text = wrapper_done.find('p');
        expect( p_text.hasClass('complete') ).toBe( true );
    })
})
