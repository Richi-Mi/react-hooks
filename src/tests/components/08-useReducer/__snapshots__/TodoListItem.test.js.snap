// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` - Pruebas en el componente <TodoListItem />  - Debe de Mostrarse correctamente  1`] = `
<li
  className="list-group-item"
  key="12234567891"
>
  <p
    className="false"
    onClick={[Function]}
  >
    1
    . - 
    Aprender a Testear
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
